src/bsoncxx/test/string_view.test.cpp

```diff
@@ -157,3 +157,48 @@ TEST_CASE("string_view: Overloading safety") {
     std::vector<string_view> vec;
     CHECK(vec == vec);
 }
+
+TEST_CASE("string_view: find") {
+    string_view s1 = "abc123abc123";
+    auto pos = s1.find("abc");
+    CHECK(pos == 0);
+    pos = s1.find("bc1");
+    CHECK(pos == 1);
+    pos = s1.find("bc1", 1);
+    CHECK(pos == 1);
+    pos = s1.find("bc1", 2);
+    CHECK(pos == 7);
+    pos = s1.find("", 4);
+    CHECK(pos == 4);
+    CHECK(s1.find("") == 0);
+    CHECK(s1.find("nowhere") == s1.npos);
+    CHECK(s1.rfind("nowhere") == s1.npos);
+    CHECK(s1.find("123") == 3);
+    CHECK(s1.rfind("123") == 9);
+    CHECK(s1.rfind("abc", 8) == 0);
```

Comment:
There appears to be behavior differences with `std::string_view`.

```c++
std::string_view s1 = "abc123abc123";
CHECK(s1.rfind("abc", 8) == 0);
```

Results in:

```c++
/Users/kevin.albertson/review/mongo-cxx-driver-1062/src/bsoncxx/test/string_view.test.cpp:223: FAILED:
  CHECK( s1.rfind("abc", 8) == 0 )
with expansion:
  6 == 0
```

Suggest running this test side-by-side with `std::string_view` to compare behavior.

Reply:
Additional tests and fixes added. Looks to be me misreading the spec and some off-by-one errors as well.